examples/advanced/vertical_xgboost/jobs/vertical_xgboost_base/app/custom/vertical_data_loader.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pandas as pd
+import xgboost as xgb
+
+from nvflare.app_opt.xgboost.data_loader import XGBDataLoader
+
+
+def _get_data_intersection(df, intersection_path, id_col):
+    with open(intersection_path) as intersection_file:
+        intersection = intersection_file.read().splitlines()
+
+    # Note: the order of the intersection must be maintained
+    intersection_df = df[df[id_col].isin(intersection)].copy()
+    intersection_df["sort"] = pd.Categorical(intersection_df[id_col], categories=intersection, ordered=True)
+    intersection_df = intersection_df.sort_values("sort")
+    intersection_df = intersection_df.drop([id_col, "sort"], axis=1)
+
+    if intersection_df.empty:
+        raise ValueError("private set intersection must not be empty")
+
+    return intersection_df
+
+
+def _split_train_val(df, train_proportion):
+    num_train = int(df.shape[0] * train_proportion)
+    train_df = df.iloc[:num_train].copy()
+    valid_df = df.iloc[num_train:].copy()
+
+    return train_df, valid_df
+
+
+class VerticalDataLoader(XGBDataLoader):
+    def __init__(self, data_split_path, label_owner, train_proportion):
+        """Reads HIGGS dataset and return data paths to train and valid sets.
```

Comment:
If this class is not limited to HIGGS dataset, suggest update the docstring to reflect it.